tests/integrationtest/l4_service_test.go

```diff
@@ -1016,3 +1016,76 @@ func TestInfraSettingChangeMapping(t *testing.T) {
 	TeardownAviInfraSetting(t, settingName2)
 	TearDownTestForSvcLB(t, g)
 }
+func TestSharedVIPSvcWithTCPUDPProtocols(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	modelName := "admin/cluster--red-ns-" + SHAREDVIPKEY
+
+	SetUpTestForSharedVIPSvcLB(t, corev1.ProtocolTCP, corev1.ProtocolUDP)
+
+	g.Eventually(func() bool {
+		found, _ := objects.SharedAviGraphLister().Get(modelName)
+		return found
+	}, 30*time.Second).Should(gomega.Equal(true))
+	_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+	nodes := aviModel.(*avinodes.AviObjectGraph).GetAviVS()
+	g.Expect(nodes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].Name).To(gomega.Equal(fmt.Sprintf("cluster--%s-%s", NAMESPACE, SHAREDVIPKEY)))
+	g.Expect(nodes[0].Tenant).To(gomega.Equal(AVINAMESPACE))
+	g.Expect(nodes[0].PortProto[0].Port).To(gomega.Equal(int32(8080)))
+	g.Expect(nodes[0].PortProto[0].Protocol).To(gomega.Equal("TCP"))
+	g.Expect(nodes[0].PortProto[1].Port).To(gomega.Equal(int32(8080)))
+	g.Expect(nodes[0].PortProto[1].Protocol).To(gomega.Equal("UDP"))
+	g.Expect(nodes[0].NetworkProfile).To(gomega.Equal(utils.MIXED_NET_PROFILE))
+	g.Expect(nodes[0].PoolRefs).To(gomega.HaveLen(2))
```

Comment:
Can we make this block into a generic method verifyProtocol(proto1, proto2 string) since repeated in other tests